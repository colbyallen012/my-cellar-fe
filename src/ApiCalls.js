export const fetchWines = () => {
  return fetch('https://the-vino-cellar.herokuapp.com/api/v1/vino')
    .then(res => res.json())
    .then(res => res)
    .catch(error => error.message)
}