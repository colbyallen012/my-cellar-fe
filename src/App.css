@import url('https://fonts.googleapis.com/css?family=Cinzel:700&display=swap');


.App {
  background-color: #282c34;
  background: linear-gradient( rgba(0, 0, 0, 0.7), rgba(0, 0, 0, 0.3) ), url("./Utils/background.jpeg");
  background-position: center;
  background-attachment: fixed;
  background-repeat: no-repeat;
  background-size: cover;
  height: 100vh;
  /* display: flex; */
  /* flex-direction: column; */
  /* align-items: center; */
  /* justify-content: center; */
  /* font-size: calc(10px + 2vmin); */
  /* color: white; */
  z-index:3;
}

h1{
  color: white;
  font-family: 'cinzel', serif;
  font-size: 65px;
  margin: 0;
  padding: 20px;
  text-align: center;
}