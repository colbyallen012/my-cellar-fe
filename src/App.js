import React, {Component} from 'react';
import { fetchWines } from './ApiCalls'
import './App.css';

export class App extends Component {
  constructor() {
    super()
    this.state = {
      wines: []
    }
  }

  componentDidMount () {
    this.getWines()
  }

  getWines = async () => {
    await fetchWines()
      .then(wines => this.setState({wines: wines}))
      .catch(error => error.message)
  }

  render () {
    return (
      <div className='App'>
        <h1>Vino Cellar</h1>
      </div>
    )
  }
}

export default App;