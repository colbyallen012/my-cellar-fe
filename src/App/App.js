import React, {Component} from 'react';
import { fetchWines, deleteWine} from '../ApiCalls/ApiCalls'
import WineForm from '../WineForm/WineForm.js'
import Wines from '../Wines/Wines.js'
import './App.css';

export class App extends Component {
  constructor() {
    super()
    this.state = {
      wines: []
    }
  }

  componentDidMount () {
    this.getWines()
  }

  getWines = async () => {
    await fetchWines()
      .then(wines => this.setState({wines: wines}))
      .catch(error => error.message)
  }

  removeVino = async (id) => {
    await deleteWine(id)
    this.getWines()
  }

  render () {
    return (
      <div className='App'>
        <h1>Vino Cellar</h1>
        <section>
          <WineForm getWines={this.getWines}/>
          <Wines wines={this.state.wines} removeVino={this.removeVino}/>
        </section>
      </div>
    )
  }
}

export default App;