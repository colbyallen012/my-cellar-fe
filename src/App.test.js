import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import { fetchWines } from './ApiCalls'
import * as constants from './ApiCalls';


describe('App', () => {
  let wrapper, instance;

  beforeEach(() => {
    wrapper = shallow(<App/>)
    instance = wrapper.instance()
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call fetchWines when getWines is called', async () => {
    constants.fetchWines = jest.fn(() => 
    Promise.resolve([]))
    expect(constants.fetchWines).toHaveBeenCalledTimes(0)
    await instance.getWines()
    expect(constants.fetchWines).toHaveBeenCalledTimes(1)
  })
})