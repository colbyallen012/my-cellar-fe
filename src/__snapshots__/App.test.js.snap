// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <h1>
    Vino Cellar
  </h1>
  <section>
    <WineForm
      getWines={[Function]}
    />
    <Wines
      removeVino={[Function]}
      wines={Array []}
    />
  </section>
</div>
`;
